rewrite-maven/src/main/java/org/openrewrite/maven/AddPlugin.java

```diff
@@ -109,6 +116,18 @@ public boolean isAcceptable(SourceFile sourceFile, ExecutionContext ctx) {
             if (filePattern != null) {
                 return PathUtils.matchesGlob(sourceFile.getSourcePath(), filePattern) && super.isAcceptable(sourceFile, ctx);
             }
+            if (!Boolean.TRUE.equals(addToRootPom)) {
+                return super.isAcceptable(sourceFile, ctx);
+            } else {
+                Optional<MavenResolutionResult> mvnResult = sourceFile.getMarkers().findFirst(MavenResolutionResult.class);
+                if (mvnResult.isPresent()) {
+                    if (mvnResult.get().getParent() == null) {
+                        return super.isAcceptable(sourceFile, ctx);
+                    } else {
+                        return false;
```

Comment:
The if block here is the same as of we were to fall through. Let's convince the two if blocks, negate the second condition, and just return false in that case to simplify the nesting here